presto-hive/src/main/java/com/facebook/presto/hive/RecordFileWriter.java

```diff
@@ -161,6 +163,8 @@ public void rollback()
     {
         try {
             recordWriter.close(true);
+            // perform explicit delete of written file here as recordWriter.close() ignore the abort flag.
+            path.getFileSystem(conf).delete(path, false);
         }
         catch (IOException e) {
```

Comment:
But `recordWriter.close()` can throw exception anywat, so if there is issue with lost resources, we have the problem already  in the current codebase.

Or do you suggest to remove the throwing of exception from `catch` clause in `rollback` at all?